File: gemini-node/whats_photo.js
const { GoogleGenerativeAI } = require("@google/generative-ai");
const fs = require("fs");
require('dotenv').config();
const GEMINI_API_KEY = process.env.GEMINI_API_KEY;

const genAI = new GoogleGenerativeAI(GEMINI_API_KEY);

async function run() {
    const prompt = "What's in this photo?";
    console.log(prompt);

    const model = genAI.getGenerativeModel({ model: "gemini-1.5-flash" });
    const result = await model.generateContent([
        prompt,
        {
            inlineData: {
                data: Buffer.from(fs.readFileSync('./images/cappuccino.webp')).toString("base64"),
                mimeType: 'image/webp'
            }
        }]
    );
    console.log(result.response.text());
}
run();

File: gemini-python/whats_photo.py
import google.generativeai as genai
import PIL.Image
import os
from dotenv import load_dotenv

# .env ファイルを読み込む
load_dotenv()

# 環境変数から API キーを取得
API_KEY = os.getenv("GEMINI_API_KEY")

# 画像読み込み
img = PIL.Image.open('images/cappuccino.webp')

# Gemini APIプロンプト
prompt = "この写真はなんですか？"

genai.configure(api_key=API_KEY)
model = genai.GenerativeModel(model_name="gemini-1.5-flash")
response = model.generate_content([prompt, img])

if (response):
    print(response.text)
else:
    print("Gemini Error")